development/install.sh

```diff
@@ -0,0 +1,383 @@
+#!/bin/bash
+
+INSTALL_APT_PKGS=
+APT_PKGS="tinyproxy
+apt-transport-https
+ca-certificates
+curl
+git
+gnupg2
+php-pear
+npm
+software-properties-common
+subversion
+yum
+"
+
+is_apt_pkg_installed() {
+  dpkg -l ${1} >>/dev/null 2>&1
+}
+
+for PKG in $APT_PKGS
+do
+  is_apt_pkg_installed ${PKG}
+  if [ $? -ne 0 ]; then
+    [ -z "${INSTALL_APT_PKGS}" ] && INSTALL_APT_PKGS="${PKG}" || INSTALL_APT_PKGS="${INSTALL_APT_PKGS} ${PKG}"
+  fi
+done
+
+if [ -n "${INSTALL_APT_PKGS}" ]; then
+  apt-get update
+  apt-get install -y ${INSTALL_APT_PKGS}
+fi
+
+command -v docker >>/dev/null
+if [ $? -ne 0 ]; then
+  curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+
+  add-apt-repository \
+     "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+     $(lsb_release -cs) \
+     stable"
+
+  apt-get update
+  apt-get -y install docker-ce
+fi
+
+cat > /etc/tinyproxy/tinyproxy.conf <<EOF
```

Comment:
Maybe this could be installed with the file provider?

Reply:
Hmm, never tried that but sure, i'll give that a shot.  I'm thinking about moving the proxy out of the test vm anyway cause it causes a chicken and egg problem.  The way it exists now, we first do a `vagrant up` without the proxy being enabled because the shell provisioner installs the proxy during the first bootstrap, then once it's bootstrapped we can use the 127.0.0.1 proxy inside the vm.  But yeah, without using something serverspec though, it may not be needed.

Reply:
> But yeah, without using something serverspec though, it may not be needed.

I dreamed about adding serverspec tests for this project already at the beginning. But never got it done. Tests require quite a lot of work in these kind of programs.

Reply:
I tried yesterday getting serverspec + test kitchen working and ran into Gem dependency problems and after a few hours of trying different things I decided it may not be worth all the hassle.

It's been a few years since I've used test kitchen, our puppetlabs beaker or just plain serverspec.  But yeah, you are right it's a bit of an undertaking to get all that setup.  Our unit tests do cover most of the code paths now however. ;)

I did briefly look at polipo-box and I accidentally ran across Tinyproxy while doing some research.  I found that tinyproxy was just install the package, do some basic configuration and start the service and watch the log file.  I didn't spend more than 10 minutes getting it all working which is nice.  I've got a few ideas id like to try today if I have the free time to make this a little easier to test.